python/ray/tune/trial_runner.py

```diff
@@ -146,6 +146,7 @@ def _process_events(self):
         del self._running[result_id]
         try:
             result = ray.get(result_id)
+            trial.result_logger.on_result(result)
```

Comment:
Do you want to check `if trial.result_logger:` before this? You do this in `trial.py`.

Reply:
This should be ok since here we know the trial is in a state where it is not None (and should raise if it is).

I removed some checks from agent since result_logger can never be None there.